src/Core/Settings.h

```diff
@@ -480,6 +480,7 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(Bool, optimize_move_functions_out_of_any, false, "Move functions out of aggregate functions 'any', 'anyLast'.", 0) \
     M(Bool, optimize_normalize_count_variants, true, "Rewrite aggregate functions that semantically equals to count() as count().", 0) \
     M(Bool, optimize_injective_functions_inside_uniq, true, "Delete injective functions of one argument inside uniq*() functions.", 0) \
+    M(Bool, optimize_countdistinctif_rewrite, false, "Rewrite countDistinctIf with count_distinct_implementation configuration", 0) \
```

Comment:
Probably better to name setting like `optimize_rewrite_count_distinct_if`.

Reply:
And I think it's better to enable it by default.